src/sql/workbench/browser/designer/designerActions.ts

```diff
@@ -0,0 +1,131 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Table } from 'sql/base/browser/ui/table/table';
+import { Designer } from 'sql/workbench/browser/designer/designer';
+import { DesignerEditType, DesignerPropertyPath, DesignerTableProperties, DesignerUIArea } from 'sql/workbench/browser/designer/interfaces';
+import { Action } from 'vs/base/common/actions';
+import { localize } from 'vs/nls';
+
+export interface DesignerTableActionContext {
+	table: Table<Slick.SlickData>;
+	path: DesignerPropertyPath;
+	source: DesignerUIArea;
+}
+
+export class AddRowAction extends Action {
+	public static ID = 'designer.addColumnAction';
+	public static ICON = 'add-row-button new codicon';
+	public static LABEL = localize('designer.addColumnAction', 'Add New');
+
+	constructor(private designer: Designer, tableProperties: DesignerTableProperties) {
+		super(AddRowAction.ID, tableProperties.labelForAddNewButton || AddRowAction.LABEL, AddRowAction.ICON);
+		this.designer = designer;
+		this._tooltip = localize('designer.newRowButtonAriaLabel', "Add new row to '{0}' table", tableProperties.ariaLabel);
+	}
+
+	public override async run(context: DesignerTableActionContext): Promise<void> {
+		Promise.resolve(() => {
+			this.designer.handleEdit({
+				type: DesignerEditType.Add,
+				path: context.path,
+				source: context.source
```

Comment:
We also need to add value to `AddRowAction` as we have index parameter for all add requests.